templates/aspnet-s2i-template.json

```diff
@@ -280,7 +277,7 @@
         {
             "name": "ASP_UPSTREAM_IMAGE_TAG",
             "description": "Base OpenShift ASP S2I Image Tag",
-            "value": "latest",
+            "value": "1.0",
```

Comment:
these are imagestreamtags, not docker image tags, so the kubernetes doc isn't really applicable.

in our imagestreams, we normally define the "latest" imagestreamtag as a virtual tag that follows a real tag.  The real tag it follows is the newest version of the image, see:
https://github.com/openshift/origin/blob/master/examples/image-streams/image-streams-centos7.json#L18-L30

For that reason, the latest tag includes a warning that if you use it, things can change underneath you in incompatible ways (we can change what latest points to at any time).

As a result of that, most of our templates do not reference the "latest" tag when they consume imagestreams, but instead reference a specific version, eg:
https://github.com/openshift/origin/blob/master/examples/db-templates/mongodb-persistent-template.json#L94-L98
https://github.com/openshift/origin/blob/master/examples/db-templates/mongodb-persistent-template.json#L270-L273

Assuming the .net imagestream is going to follow that same pattern of having the latest tag be a virtual tag, it would make sense for .net templates that depend on the .net imagestream, to reference a specific version tag, not latest.

But it has nothing to do w/ whether the image will be pulled or not, it's all about whether you want to stay on a compatible image version, or always get the latest version of the language/framework/db available in the cluster.